crates/optimism/consensus/src/validation/mod.rs

```diff
@@ -48,31 +47,13 @@ where
         ))
     }
 
```

Comment:
up here we need to check that body withdrawals are empty with the canyon function

Reply:
in my perspective, instead of duplicating the conditional logic across `mod.rs`, better to consider abstracting the withdrawal-related checks into a shared helper (maybe under `consensus::validation::withdrawals`)?
This would eliminate inconsistencies like the `else` vs `else if canyon.is_active` case, and make it clearer which forks are responsible for which checks.
It also helps to avoid future regressions around fork-specific assumptions.